src/scalib_ext/scalib/src/sasca/factor_graph.rs

```diff
@@ -323,7 +326,16 @@ impl FactorGraph {
                             .unwrap()
                             .map(|x| self.tables[*table].values[x as usize]),
                     };
-                    if &res != expected_res {
+                    let check = match (&res, expected_res) {
```

Comment:
Looks ok.

This kind of weird stuff smells like the code above is not really well architectured, and should probably ensure that `res` always matches `expected_res` w.r.t. Single/Multi.